rules/common/private/utils.bzl

```diff
@@ -42,6 +42,19 @@ def _format_classpath_entry(runfiles_enabled, workspace_prefix, file):
 
     return "$(rlocation " + paths.normalize(workspace_prefix + file.short_path) + ")"
 
+def _format_javabin(java_executable, workspace_prefix, runfiles_enabled):
+    if not paths.is_absolute(java_executable):
+        java_executable = workspace_prefix + java_executable
+    java_executable = paths.normalize(java_executable)
+
+    if runfiles_enabled:
+        prefix = "" if paths.is_absolute(java_executable) else "${JAVA_RUNFILES}/"
+        javabin = "JAVABIN=${JAVABIN:-" + prefix + java_executable + "}"
```

Comment:
Nit: Why not just return this directly?

Reply:
No good reason. I'll change that. I forget that starlark has return statements like that.